q2_assembly/mason/mason.py

```diff
@@ -0,0 +1,289 @@
+
+import numpy as np
+import pandas as pd
+from q2_types.genome_data import GenomeSequencesDirectoryFormat
+from q2_types.per_sample_sequences import CasavaOneEightSingleLanePerSampleDirFmt
+
+from .._utils import run_command
+
+
+def _process_mason_arg(arg_key, arg_val):
+    """Creates a list with argument and its value.
+
+    Argument values represented by a list will be converted to a single
+    string joined by spaces, e.g.: [1, 2, 3] -> '1 2 3'.
+    Argument names will be converted to command line parameters by
+    appending a '--' prefix, e.g.: 'some_parameter' -> '--some_parameter'.
+
+    Args:
+        arg_key (str): Argument name.
+        arg_val: Argument value.
+
+    Returns:
+        [converted_arg, arg_value, ...]: List containing a prepared command
+            line parameter and its value(s).
+    """
+    if isinstance(arg_val, bool) and arg_val:
+        return [f"--{arg_key}"]
+    elif not isinstance(arg_val, list):
+        return [f"--{arg_key}", str(arg_val)]
+    else:
+        flags = [f"--{arg_key}"]
+        flags.extend([str(x) for x in arg_val])
+        return flags
+
+
+def generate_abundances(
+    profiles, num_genomes, mu=0, sigma=1, lambd=0.5, random_seed=42
+):
+    np.random.seed(random_seed)
+    all_abundances = []
+    for profile in profiles:
+        abundances = []
+        if profile == "uniform":
+            abundances = [1 / num_genomes] * num_genomes
+        elif profile == "exponential":
+            abundances = np.exp(-lambd * np.arange(num_genomes))
+            abundances /= abundances.sum()
+        elif profile == "lognormal":
+            abundances = np.exp(mu + sigma * np.random.randn(num_genomes))
+            abundances /= abundances.sum()
+        else:
+            print(f"Invalid abundance profile option: {profile}")
+        all_abundances.append(list(abundances))
+    return all_abundances
+
+
+def abundances_to_df(abundances, genome_files, sample_id):
+    data = {
+        "id": [os.path.basename(f).split(".")[0] for f in genome_files],
+        sample_id: abundances,
+    }
+    df = pd.DataFrame(data)
+    df.set_index("id", inplace=True)
+    return df
+
+
+def _combine_reads(sample_id, src_dir, orientation="forward"):
+    _idx = "1" if orientation == "forward" else "2"
+
+    all_files = sorted(
+        glob.glob(os.path.join(src_dir, f"{sample_id}_*_00_L001_R{_idx}_001.fastq.gz"))
+    )
+    with open(
+        os.path.join(src_dir, f"{sample_id}_00_L001_R{_idx}_001.fastq.gz"), "wb"
+    ) as out:
+        cmd = ["cat", *all_files]
+        subprocess.run(cmd, check=True, stdout=out)
+
+    # clean up
+    for f in all_files:
+        os.remove(f)
+
+
+def _process_sample(
+    sample, genome_files, abundances, total_reads, tmp_dir, threads, read_len, seed
+):
+    for genome_file, abundance in zip(genome_files, abundances):
+        genome_reads = int(total_reads * abundance)
+        _id = os.path.basename(genome_file).replace(".fasta", "")
```

Comment:
```suggestion
        _id = os.path.splitext(os.path.basename(genome_file))[0]
```
The FASTA files in the GenomeSequencesDirectoryFormat can also have the fa extension.